contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,190 @@
+     - Generate Angular files - `ng generate`
+     - Build our application for deployment - `ng build`
+
+This [article](https://medium.freecodecamp.org/best-practices-for-a-clean-and-performant-angular-application-288e7b39eb6f) provides a comprehensive list of best tooling services offered by Angular.
+ 
+### Benefit 2: Angular allows Cross-platform development.
+
+Using the modern versions of Angular, developers can build applications that live on the web, mobile, or the desktop.
+
+* **Progressive Web Apps** - PWAs are web-based applications that contain most of the benefits of a typical mobile app. The [official documentation](https://developers.google.com/web/progressive-web-apps/) by Google describes PWAs to be reliable, fast and engaging. Given it's "best practices", Angular is a good choice to build PWAs.
+This [article](https://medium.com/@nsmirnova/creating-pwa-with-angular-5-part-2-progressifying-the-application-449e3a706129) is a good place to learn the basics of building PWAs using Angular.
+
+* **Native Apps** - Angular can be used to build <tooltip content="a smartphone application that is coded in a specific programming language, such as Swift for iOS or Java for Android operating systems">native mobile applications</tooltip>. There are many frameworks such as [NativeScript](https://www.nativescript.org/) and [Ionic](https://ionicframework.com/)
+that can be integrated with Angular for mobile application development. The [official Ionic website](https://ionicframework.com/docs/intro) claims that "integration with the Angular ecosystem is a breeze".
+
+* **Desktop Apps** - With Angular you can create desktop applications across Mac, Windows and Linux systems. A common approach used by many developers is to use [Electron](https://electronjs.org/), a framework maintained by Github. You can install the Electron framework in your Angular project with the help of the CLI and start building cross-platform applications.
+
+### Benefit 3: Angular provides two-way data binding.
+
+As explained previously, Angular divides the application into the Model, View and Controller architecture. Data binding establishes the connection between the Model and View components. There are two types of data binding as explained below.
+
+![Data Binding](one_two.png)
+
+Angular uses two-way data binding as opposed to popular tools like React that use one way data binding. How does this help?
+
+It ensures the exchange of data between the View and Model in a bi-directional manner. If the User Input changes, data in the backend is updated by Angular. Additionally, if the Model state (data) changes, the UI is updated automatically. This means reduced coding for developers.
+
+One-way data binding (used in React) involves uni-directional communication. If the model is changed, the changes are reflected in the UI. However, if the UI element changes the model state is not updated. Developers must provide this synchronization code. 
```

Comment:
Perhaps reorganize and simplify the explanation along these lines?
* In web apps, there is a view, and there is a model. view is ..., model is ...
* Some frameworks e.g., React provides one-way data binding i.e., when there is a change to the model, view updates automatically. If the app allows users to change data thorough the view (e.g., ...), the developer has to write code to propagate that change to the model.
* In contrast, Angular provides data binding in both ways. That means even changes to the view can be propagated to the model automatically, without writing extra code for that.